torch/csrc/jit/passes/quantization.cpp

```diff
@@ -504,6 +515,62 @@ void InsertQuantDequantNodesForParam(
   }
 }
 
+// Adding quant-dequant node for bias is considered as special case because
+// qparam for bias depend on scale factors for input activation and weight.
+void InsertQuantDequantNodesForBias(
+    script::Method& method,
+    const std::function<std::tuple<std::string, float, int>(float, float)>&
+        getQParamFunc) {
+  AT_ASSERT(getQParamFunc != nullptr);
+  auto params_to_insert_qdq =
+      getQuantizableParamsofType(method, std::string("bias"));
+
+  for (auto& param_info : params_to_insert_qdq) {
+    // Fetch scale for input activation and weight to the quantizable
+    // node for which bias need to be quantized
+    Node* n = param_info.n;
+    // Check if this node has weight attr as input
+    size_t param_index = getParamIndexinOpArgs(n, std::string("weight"));
+    if (param_index >= n->inputs().size()) {
+      // No attribute by name weight
+      continue;
+    }
+    std::vector<size_t> node_inputs_idx{0, param_index};
+    std::array<float, 2> scale_factors;
+    bool skip_node = false;
+    for (size_t idx = 0; idx < node_inputs_idx.size(); idx++) {
+      size_t input_index = node_inputs_idx[idx];
+      Value* input_value = n->inputs()[input_index];
+      Node* n_input_value = input_value->node();
+      if (n_input_value->kind().toQualString() !=
+              std::string("aten::dequantize_linear") ||
+          !n_input_value->inputs().size()) {
+        // Dequant node pattern for input is missing
+        skip_node = true;
+        break;
+      }
+      float input_scale = getScaleforNode(n_input_value);
+      AT_ASSERT(input_scale != 0.0);
+      scale_factors[idx] = input_scale;
+    }
+    if (skip_node) {
+      continue;
+    }
+    auto bias_qparam = getQParamFunc(scale_factors[0], scale_factors[1]);
+    addQuantDeQuantNodesForInput(
+        param_info.v, param_info.n, bias_qparam, at::ScalarType::Int);
```

Comment:
ScalarType::Int -> ScalarType::QInt32 after my stack of diff lands

Reply:
done